src/Lecture1.hs

```diff
@@ -108,7 +128,20 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str =
+    let l = words str
+    in sumStrList l
+
+sumStrList :: [String] -> Int
+sumStrList l =
+    if null l 
+    then 0
+    else    
+        let
+            s = read (head l) :: Int
```

Comment:
I recommend using pattern matching instead of partial unsafe functions :) you can use `case l of` instead of `if` statement

Reply:
I did refactoring and  implemented it with foldr. Defenetly agree about partial functions. They should be avoid if it possible.